cmd/operator-sdk/cli/legacy.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/bundle"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/cleanup"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/completion"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/execentrypoint"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/generate"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/migrate"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/new"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/olm"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/printdeps"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/run"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/scorecard"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/test"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/version"
+	"github.com/operator-framework/operator-sdk/internal/flags"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func RunLegacy() error {
+	root := GetCLIRoot()
+	root.PersistentFlags().Bool(flags.VerboseOpt, false, "Enable verbose logging")
+	if err := viper.BindPFlags(root.PersistentFlags()); err != nil {
+		log.Fatalf("Failed to bind root flags: %v", err)
+	}
+
+	return root.Execute()
+}
+
+// GetCLIRoot is intended to create the base command structure for the OSDK for use in CLI and documentation
+// Deprecated: this is only used for projects without a PROJECT config/kubebuilder-style repo.
+func GetCLIRoot() *cobra.Command {
+	root := &cobra.Command{
+		Use:   "operator-sdk",
+		Short: "An SDK for building operators with ease",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if viper.GetBool(flags.VerboseOpt) {
+				if err := projutil.SetGoVerbose(); err != nil {
+					log.Fatalf("Could not set GOFLAGS: (%v)", err)
+				}
+				log.SetLevel(log.DebugLevel)
+				log.Debug("Debug logging is set")
+			}
+			if err := checkGoModulesForCmd(cmd); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+
+	root.AddCommand(
+		add.NewCmd(),
+		alpha.NewCmd(),
+		build.NewCmd(),
+		bundle.NewCmd(),
+		cleanup.NewCmd(),
+		completion.NewCmd(),
+		execentrypoint.NewCmd(),
+		generate.NewCmd(),
+		migrate.NewCmd(),
+		new.NewCmd(),
+		olm.NewCmd(),
+		printdeps.NewCmd(),
+		run.NewCmd(),
+		scorecard.NewCmd(),
+		test.NewCmd(),
+		version.NewCmd(),
+	)
+
+	// The "init" cmd is is added as a hidden subcommand from the new KB CLI to provide
+	// a way to initialize Kubebuilder projects from the legacy CLI.
+	// TODO(hasbro17): This should be removed once the KB CLI is the default.
+	initCmd := getInitCmd()
+	if initCmd == nil {
+		log.Debug("Failed to add init cmd to CLI")
+	} else {
+		initCmd.Hidden = true
+		root.AddCommand(initCmd)
+	}
+
+	return root
+}
+
+func getInitCmd() *cobra.Command {
```

Comment:
Non-blocking note: the way `init` is retrieved prevents global flags set by root, which contain `--plugins`, from being retrieved. This is not blocking because right now only one plugin is available. Once https://github.com/kubernetes-sigs/kubebuilder/pull/1465 merges, `--plugins` will be local to `init`, so we can bump kubebuilder commits to get `--plugins` registration.